code/modules/mining/machine_processing.dm

```diff
@@ -5,6 +5,33 @@
 /obj/machinery/mineral
 	var/input_dir = NORTH
 	var/output_dir = SOUTH
+	var/turf/input_turf = null // The turf that the machines will be listening to for ores/items
+	var/needs_item_input = FALSE // whether or not the machine needs to pick up items, i.e. the ORM picking up ores
+	processing_flags = START_PROCESSING_MANUALLY
+	subsystem_type = /datum/controller/subsystem/processing/fastprocess
+
+/obj/machinery/mineral/Initialize(mapload)
+	. = ..()
+	if(needs_item_input)
+		register_input_turf()
+
+/obj/machinery/mineral/Destroy()
+	. = ..()
+	if(input_turf)
+		unregister_input_turf()
+
+/obj/machinery/mineral/proc/register_input_turf()
+	input_turf = get_step(src, input_dir) // get the turf that players need to place the ore/items onto, defaults to NORTH at round start
+	if(input_turf && istype(input_turf)) // make sure there is actually a turf
+		// listens for when an atom ENTERS the input_turf, tells the machine to deal with the ores/items
+		RegisterSignal(input_turf, COMSIG_ATOM_ENTERED, .proc/pickup_items, TRUE)
+
+/obj/machinery/mineral/proc/unregister_input_turf()
+	if(input_turf)
+		UnregisterSignal(input_turf, COMSIG_ATOM_ENTERED)
+
+/obj/machinery/mineral/proc/pickup_items()

```

Comment:
Ditto, also you havnt added the parameters which a signal proc should have. Look at another proc linked to this signal for examples

Reply:
Ah yeah, I know the params you're speaking of. I'm curious why you actually need them if the proc gets executed just fine without them though.

Reply:
clarity, and in my opinion, you should be using it, it makes it far more event-based rather than looping through the input turf.

Reply:
yeah nevermind I get what you mean now, if Im not looping, I need the signal params, in particular the atom/movable :+1: